Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProject/Screens/CourseListVC.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  CourseListViewController.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 10/16/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import UIKit
+
+class CourseListVC: UIViewController {
+  private var courseContentUrl: String?
+  private var courses = [Item]()
+  private var dataSource: UITableViewDiffableDataSource<Section, Item>?
+  private let courseManager = CourseAPIService()
+  private let timeFormatter: DateComponentsFormatter = {
+    let timeFormatter = DateComponentsFormatter()
+    timeFormatter.allowedUnits = [.hour, .minute]
+    timeFormatter.unitsStyle = .short
+    return timeFormatter
+  }()
+  @IBOutlet var courseList: UITableView!
+  @IBAction func filterCourses(_ sender: UISegmentedControl) {
+    var filteredContent = [Item]()
+
+    switch sender.selectedSegmentIndex {
+    case 0:
+      filteredContent = courses
+    case 1:
+      filteredContent = courses.filter { $0.attributes.contentType == ContentType.article }
+    case 2:
+      filteredContent = courses.filter { $0.attributes.contentType == ContentType.collection }
+    default:
+      filteredContent = courses
```

Comment:
One way to make this a little nicer is to define an enum with an `Int` raw value for the sections.

```swift
enum Filter: Int {
  case all = 0, article, collection
}

switch Filter(rawValue: section) {
  case .all:
    ...
  case .article:
   ...
}
```

Reply:
fixed